### ISSUE TITLE
Duplicated key error while updating a gene panel

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#3052]
```
2021-12-13 15:55:18,956 - scout.server.app - ERROR: Exception on /panels/6033cc76852c78381b768e55/update [POST] [in /home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/blueprints/panels/views.py", line 127, in panel_update
    new_panel_id = store.apply_pending(panel_obj, update_version)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/adapter/mongo/panel.py", line 516, in apply_pending
    inserted_id = self.panel_collection.insert_one(new_panel).inserted_id
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 701, in insert_one
    session=session),
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 615, in _insert
    bypass_doc_val, session)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 603, in _insert_one
    acknowledged, _insert_command, session)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1498, in _retryable_write
    return self._retry_with_session(retryable, func, s, None)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1384, in _retry_with_session
    return self._retry_internal(retryable, func, session, bulk)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1416, in _retry_internal
    return func(session, sock_info, retryable)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 600, in _insert_command
    _check_write_command_response(result)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/helpers.py", line 226, in _check_write_command_response
    _raise_last_write_error(write_errors)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/helpers.py", line 207, in _raise_last_write_error
    raise DuplicateKeyError(error.get("errmsg"), 11000, error)
pymongo.errors.DuplicateKeyError: E11000 duplicate key error collection: scout.gene_panel index: panel_name_1_version_1 dup key: { panel_name: "IEM", version: 18.0 }, full error: {'index': 0, 'code': 11000, 'keyPattern': {'panel_name': 1, 'version': 1}, 'keyValue': {'panel_name': 'IEM', 'version': 18.0}, 'errmsg': 'E11000 duplicate key error collection: scout.gene_panel index: panel_name_1_version_1 dup key: { panel_name: "IEM", version: 18.0 }'}
```

**Comment by northwestwitch** [comment]
I've been trying to reproduce this error but so far I couldn't. I've tried updating a panel while keeping the same version ID: it doesn't crash. Tried also using a database version (4.4) different from the one I have locally, still doesn't crash. Tried sending exactly the same request for updating a panel from two different windows (one after the other). Still doesn't crash. I guess we need to ask the user how they triggered the error. Let's leave it for now, since they say it was not a big deal and the panel update worked fine after the first attempt!
